app/assets/javascripts/poller.js

```diff
@@ -36,7 +37,11 @@
     },
 
     insertActivity: function(text) {
-      $(text).hide().prependTo(this.$poller).fadeIn();
+      var element = $(text);
+
+      if (this.$poller.find('#' + element.attr('id')).length == 0) {
```

Comment:
As id's are unique to the whole page you could probably just use `getElementById` and then boolean-ise it. Does this seem readable?

```js
if (!!!document.getElementById(element.attr('id')) {
```